src/zamp_collector/zcl_amp_c_batch_input.clas.abap

```diff
@@ -35,19 +35,42 @@ CLASS zcl_amp_c_batch_input IMPLEMENTATION.
     metrics_current_run = zcl_amp_collector_utils=>initialize_metrics( metrics_last_run = metrics_last_run ).
 
     SELECT
-    COUNT(*) AS count,
-    groupid AS groupid,
-    progid AS progid,
-    qstate AS state
-    FROM apqi
-    WHERE ( datatyp = 'BDC' OR datatyp = 'BDCE' OR datatyp = 'RODC' )
-    GROUP BY groupid, progid, qstate
-    INTO TABLE @batch_inputs.
+      COUNT(*) AS count,
+      groupid AS groupid,
+      progid AS progid,
+      qstate AS state
+      FROM apqi
+      WHERE ( datatyp = 'BDC' OR datatyp = 'BDCE' OR datatyp = 'RODC' )
+      GROUP BY groupid, progid, qstate
+      INTO TABLE @batch_inputs.
+
+    SELECT
```

Comment:
Maybe this SELECT could be replaced by a calculation of the first SELECT., but it's not a problem :)

Reply:
Also possible 
could be changed